### PR TITLE
Add TheJobOverflow Problem Parser

### LINKED ISSUE #356: Extension not working on TheJobOverflow Website
Extension is not working on [TheJobOverflow](http://thejoboverflow.com/problems) Website.

### PR DESCRIPTION
**Pull request by touhidurrr**
resolves #356 

I tested and it works.

### CODE REVIEW

File: src/parsers/problem/TheJobOverflowProblemParser.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  public async parse(url: string, html: string): Promise<Sendable> {
+    const doc = htmlToElement(html);
+    const task = new TaskBuilder('TheJobOverflow').setUrl(url);
+
+    const titleStr = doc.querySelector('#nav-problem h5').textContent;
+    task.setName(titleStr.replace(/\s*:$/, ''));
+
+    const inputs = [...doc.querySelectorAll('.sample-test > .input > .content')];
+    const outputs = [...doc.querySelectorAll('.sample-test > .output > .content')]
+    for (let i = 0; i < inputs.length; i++) {
```
Comment (jmerle) [inline src/parsers/problem/TheJobOverflowProblemParser.ts:20]: Similar to your other PR, the constraint, you use `outputs[i]` in the body so `i < inputs.length` should be `i < input.length && i < outputs.length`.

**Review by jmerle — CHANGES_REQUESTED** [review]
Looking good, just two minor comments.

File: src/parsers/problem/TheJobOverflowProblemParser.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import { Sendable } from '../../models/Sendable';
+import { TaskBuilder } from '../../models/TaskBuilder';
+import { htmlToElement } from '../../utils/dom';
+import { Parser } from '../Parser';
+
+export class TheJobOverflowProblemParser extends Parser {
+  public getMatchPatterns(): string[] {
+    return ['http://thejoboverflow.com/problem/*'];
```
Comment (jmerle) [inline src/parsers/problem/TheJobOverflowProblemParser.ts:8]: TheJobOverflow also seems to support https, please include "https://thejoboverflow.com/problem/*" as well.

**Review by jmerle — CHANGES_REQUESTED** [review]
Sorry, I forgot to approve the GitHub Actions build yesterday, please run `pnpm fix` to fix code style issues and make the build pass. Other than that it looks good!

**Review by jmerle — APPROVED** [review]
LGTM :+1:

Will be released after #358 is merged.

**Comment by sagnikghoshcr7** [comment]
Still it is not working

**Comment by touhidurrr** [comment]
> working

Wait for Chrome Web Store to show version 2.4.4

**Comment by touhidurrr** [comment]
@jmerle how much time does it take for the extension update to be available on Chrome Web Store?

**Comment by touhidurrr** [comment]
@sagnikghoshcr7 is it working now?

**Comment by sagnikghoshcr7** [comment]
Yes

**Comment by sagnikghoshcr7** [comment]
@touhidurrr Yes

**Comment by jmerle** [comment]
> @jmerle how much time does it take for the extension update to be available on Chrome Web Store?

When I submit a new version to the Chrome Web Store it says it can take up to 30 days for it to be reviewed, but usually it happens within 24 hours. I don't know at what interval Chrome checks for extension updates, that probably adds another ~24 hours to the process of getting the latest version.

**Pull request merged**